lib/ModelRoot.js

```diff
@@ -28,11 +28,14 @@ function ModelRoot(o) {
 ModelRoot.prototype.errorSelector = function errorSelector(x, y) {
     return y;
 };
-ModelRoot.prototype.comparator = function comparator(a, b) {
-    if (hasOwn(a, "value") && hasOwn(b, "value")) {
-        return a.value === b.value;
+ModelRoot.prototype.comparator = function comparator(cacheNode, messageNode) {
+    if (hasOwn(cacheNode, "value") && hasOwn(messageNode, "value")) {
```

Comment:
I also was going to do the same thing.  

* There is a major performance concern.  I can quantify the numbers but it will no be cheap.  
* Metadata outside our domain knowledge should probably be up to the user to compare.  They could provide their own comparator if they need a new/additional set of rules.

Imagine this
```javascript
var a = {
    $type: 'atom',
    $meta: { lots: 'of', fields: 'here'},
    value: 6
}
var b = deepCopy(a);
comparator(a, b) // with all meta keys compared, this would be false, since object comparison is always false if they are not the same object.
```